include/ssm.h

```diff
@@ -663,6 +663,39 @@ ssm_value_t ssm_new_adt(uint8_t val_count, uint8_t tag);
 
 /** @} */
 
+
+/** HANS TODO: closure docs **/
+
+typedef ssm_act_t *(*ssm_func_t)(ssm_act_t *parent, ssm_priority_t prio,
+                                ssm_depth_t depth, ssm_value_t *argv,
+                                ssm_value_t *ret);
+
+struct ssm_closure1 {
+  struct ssm_mm mm;
+  ssm_func_t f;
+  ssm_value_t argv[1];
+};
+
+ssm_value_t ssm_new_closure(ssm_func_t f);
+
+#define ssm_closure_arg(v, i)	\
+  (&*container_of((v).heap_ptr, struct ssm_closure1, mm)->argv)[i]
+
+#define ssm_closure_argv(v)	\
```

Comment:
Can you put `&*` in front of this to cast away the array size?

Reply:
Does 540511b1c resolve this? Not 100% sure if I understood you correctly.

Reply:
Yeah that looks right to me